File: src/com/Torvald/Terrarum/ConsoleCommand/Zoom.java
package com.Torvald.Terrarum.ConsoleCommand;

import com.Torvald.Terrarum.Game;
import com.Torvald.Terrarum.Terrarum;

/**
 * Created by minjaesong on 16-01-25.
 */
public class Zoom implements ConsoleCommand {
    @Override
    public void execute(String[] args) {
        if (args.length == 2) {

            float zoom;
            try {
                zoom = new Float(args[1]);
            }
            catch (NumberFormatException e) {
                new Echo().execute("Wrong number input.");
                return;
            }

            if (zoom < Terrarum.game.ZOOM_MIN) {
                zoom = Terrarum.game.ZOOM_MIN;
            }
            else if (zoom > Terrarum.game.ZOOM_MAX) {
                zoom = Terrarum.game.ZOOM_MAX;
            }

            Terrarum.game.screenZoom = zoom;

            System.gc();

            new Echo().execute("Set screen zoom to " + String.valueOf(zoom));
        }
        else {
            printUsage();
        }
    }

    @Override
    public void printUsage() {
        new Echo().execute("Usage: zoom [zoom]");
    }
}

File: src/net/torvald/terrarum/modulebasegame/console/Zoom.kt
package net.torvald.terrarum.modulebasegame.console

import net.torvald.terrarum.Terrarum
import net.torvald.terrarum.console.ConsoleCommand
import net.torvald.terrarum.console.Echo

/**
 * Created by minjaesong on 2016-01-25.
 */
internal object Zoom : ConsoleCommand {
    override fun execute(args: Array<String>) {
        if (args.size == 2) {

            var zoom: Float
            try {
                zoom = args[1].toFloat()
            }
            catch (e: NumberFormatException) {
                Echo("Wrong number input.")
                return
            }

            if (zoom < Terrarum.ingame!!.ZOOM_MINIMUM) {
                zoom = Terrarum.ingame!!.ZOOM_MINIMUM
            }
            else if (zoom > Terrarum.ingame!!.ZOOM_MAXIMUM) {
                zoom = Terrarum.ingame!!.ZOOM_MAXIMUM
            }

            Terrarum.ingame!!.screenZoom = zoom

            System.gc()

            Echo("Set screen zoom to " + zoom.toString())
        }
        else {
            printUsage()
        }
    }

    override fun printUsage() {
        Echo("Usage: zoom [zoom]")
    }
}
